### PR TITLE
ASC-705 Add unit tests for get_id_by_name

### PR DESCRIPTION
**Pull request by johnduarte**
This commit adds unit tests for the `get_id_by_name` helper.

### CODE REVIEW

**Comment by johnduarte** [comment]
Thanks @zreichert I have added the json library and used it to create the json strings from dictionaries.

File: tests/helpers/test_get_id_by_name.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+def test_invalid_json(mocker):
+    """Verify get_id_by_name returns None when OpenStack query returns invalid
+    JSON."""
+
+    myout = mocker.MagicMock(stdout='')
+    myhost = mocker.MagicMock()
+    myhost.run.return_value = myout
+
+    result = pytest_rpc.helpers.get_id_by_name('server', 'myserver', myhost)
+    assert result == None  # noqa
```
Comment (ryan-rs) [inline tests/helpers/test_get_id_by_name.py]: The evaluation should be the following when checking to see if something is None:

```python
assert result is None
```

File: tests/helpers/test_get_id_by_name.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    """Verify get_id_by_name returns None when OpenStack query returns JSON
+    without an id value."""
+
+    out = json.dumps({'foo': 'bar'})
+    myout = mocker.MagicMock(stdout=out)
+    myhost = mocker.MagicMock()
+    myhost.run.return_value = myout
+
+    result = pytest_rpc.helpers.get_id_by_name('server', 'myserver', myhost)
+    assert result == None  # noqa
```
Comment (ryan-rs) [inline tests/helpers/test_get_id_by_name.py]: The evaluation should be the following when checking to see if something is None:

```python
assert result is None
```

**Comment by johnduarte** [comment]
Thanks @ryan-rs ! Assertions updated.

**Pull request merged**